### PR TITLE
feat(governance): proposal list tile and summary enhancements

### LINKED ISSUE #3939: Proposal list tile and summary enhancements
## Story

As a user with a keen eye for community governance but potentially limited time to review all proposals in detail 
I want to be able to see the key details of a proposal and its status on the proposal list view
So that I can more readily assess the broad state of governance on the vega network at a glance

## Acceptance Criteria

On the Proposals list 
- [ ] I can see the closing date and time on the tile 
- [ ] I can see how long is left to vote (might do this) 
- [ ] I can see colourful / shouty things that prompt me to vote urgently on things "CLOSING SOON" 
- [ ] I can see the current voting status and whether it will pass 
- [ ] I can see the current participation and whether it will pass

On the Homepage
- [ ] I can see the top most recent 4 proposals with the same information as the cards on the proposals listing

On opening a proposal, I can see a summary of the most important information for a proposal first 
- [ ] I can see the proposal title [ rationale / title ] 
- [ ] I can see the market code [ instrument / code ]
- [ ] I can see whether majority has been reached for or against 
- [ ] I can see if participation levels have been reached to close the vote 
- [ ] I can see when the vote will end 

When my wallet is connected 
- [ ] I can see if I have voted for / against immediately on opening a proposal 
- [ ] I can see whether I have already voted or not on the card / list view (s) 

## Tasks

- [ ] Explore and sketch 
- [ ] Team and stakeholder review
- [ ] Design in context of whole page, considering the ID, Closed on, Enacted on, Proposed by, Proposed on summary placement as well as the rest of the page
- [ ] Team review
- [ ] Etc.

## Designs 
Homepage and proposal list cards: https://www.figma.com/file/X9l0t8TZUF1tU8u0ILV3ws/Governance?type=design&node-id=2838-20488&mode=design&t=YxXEdT3jxszC76M5-0

Image

Proposal details page (summary at the top): https://www.figma.com/file/X9l0t8TZUF1tU8u0ILV3ws/Governance?type=design&node-id=2846-23539&mode=design&t=3nQAyhFgGK4tDRCu-0

Image

## Sketches

Image

Image

Image

### PR DESCRIPTION
**Pull request by sam-keen**
# Related issues 🔗

Closes #3939

Screenshot 2023-07-14 at 15 14 45

Screenshot 2023-07-13 at 16 57 25

Screenshot 2023-07-14 at 09 48 54

### CODE REVIEW

File: apps/governance/src/routes/proposals/test-helpers/mocks.ts [hunk 1]
```diff
@@ -62,3 +75,34 @@ export const lastWeek = new Date(-oneWeek);
 export const nextWeek = new Date(oneWeek);
 export const lastMonth = new Date(-oneMonth);
 export const nextMonth = new Date(oneMonth);
+
+export const createUserVoteQueryMock = (
+  proposalId: string | undefined | null,
+  value: VoteValue
+) => ({
```
Comment (mattrussell36) [inline apps/governance/src/routes/proposals/test-helpers/mocks.ts:82]: Its important to type these as if you miss a field the mock query will not resolve

```suggestion
): MockedResponse<GeneratedVoteQuery> => ({
```

File: libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-vote.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export const IconVote = ({ size = 16 }: { size: number }) => {
+  return (
+    <svg width={size} height={size} viewBox="0 0 24 24">
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-vote.tsx:3]: Can you clean this icon up? I doubt the fillRule and clipRule are needed. Also all viewboxes for these icons should be 0 0 16 16 so we can get consistent behavior when sizing them.

File: apps/governance/src/routes/proposals/components/proposal-voting-status/proposal-voting-status.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+          networkParams.governance_proposal_updateNetParam_requiredMajority;
+        break;
+      case 'NewFreeform':
+        requiredVotingMajority =
+          networkParams.governance_proposal_freeform_requiredMajority;
+        break;
+    }
+  }
+
+  return (
```
Comment (edd) [inline apps/governance/src/routes/proposals/components/proposal-voting-status/proposal-voting-status.tsx:135]: **nitpick (non-blocking)**: I don't like this whole return block being an and statement. On the other hand, it's not hard to follow so it's not really a problem at all. I just had to put an opinion in ¯\_(ツ)_/¯

Comment (mattrussell36) [reply apps/governance/src/routes/proposals/components/proposal-voting-status/proposal-voting-status.tsx:135]: Agree, would be more readable as

```
if (isUpdateMarket) {
  return (
    <div className="mb-6">
        <p>{t('Token vote')}</p>
}

return (
  <div className="grid grid-cols-2 gap-4 items-center mb-6">
        <MajorityStatus
        ...
)
``

File: apps/governance/src/routes/proposals/components/proposal-voting-status/proposal-voting-status.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+          <ProposalVotingStatus
+            proposal={proposal}
+            networkParams={mockNetworkParams}
+          />
+        </VegaWalletContext.Provider>
+      </MockedProvider>
+    </Router>
+  );
+
+beforeAll(() => {
```
[the remark this answers is not in the document]
Comment (sam-keen) [reply apps/governance/src/routes/proposals/components/proposal-voting-status/proposal-voting-status.spec.tsx:62]: It's still scoped to the file, but for the sake of convention I'll move it in

File: libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-vote.tsx (hunk 2 above)
Comment (sam-keen) [reply libs/ui-toolkit/src/components/icon/vega-icons/svg-icons/icon-vote.tsx:3]: The fillrule and cliprule are needed for the internal 'tick' inside the icon. I'll resize it if it bothers you, but it'll make no different - the viewbox only determines the relative size for the coordinates. The 'width' and 'height' properties are what will determine the rendered size. Also note there are other icons in the set that aren't set to a viewbox of the same size.

File: apps/governance/src/routes/proposals/components/proposal-voting-status/proposal-voting-status.tsx (hunk 3 above)
Comment (sam-keen) [reply apps/governance/src/routes/proposals/components/proposal-voting-status/proposal-voting-status.tsx:135]: I find that less readable, but fair enough

**Review by mattrussell36 — APPROVED** [review]
